ui/src/components/ConfigurationForm/ConfigurationForm.component.tsx

```diff
@@ -62,6 +69,23 @@ export const ConfigurationForm: React.FC<ConfigurationFormProps> = ({ isEmbedded
     },
   });
 
+  const handleShowPreview = useCallback(() => {
+    // 'trigger()' triggers the validation.
+    form.trigger().then(isValid => {
+      if (isValid) {
+        const casted = form.getValues() as StarterRequest;
+        // Conversion of bools is done by hand, because casting writes booleans as strings.
+        const formData: StarterRequest = {
+          ...casted,
+          addDocumentation: 'true' === casted.addDocumentation.toString(),
```

Comment:
tip: you can do `JSON.parse(casted.addDocumentation.toString())` to get the boolean `true` from a string if you're sure that it will always be either `'true'` or `'false'`

not sure about the readability of this approach tho :)